airflow/jobs.py

```diff
@@ -668,8 +706,8 @@ def _execute(self):
         dagbag = models.DagBag(self.subdir, sync_to_db=True)
         executor = self.executor = dagbag.executor
         executor.start()
-        i = 0
-        while not self.num_runs or self.num_runs > i:
+        self.runs = 0
```

Comment:
Yes, that it actually counts :-).  When I kept it a local variable it would not properly increase. I am not sure why.